src/sentry/api/endpoints/project_environment_details.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import absolute_import
+
+from rest_framework import serializers
+from rest_framework.response import Response
+
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import EnvironmentProject
+
+
+class ProjectEnvironmentSerializer(serializers.Serializer):
```

Comment:
Name isn't mutable since this is a join table between `Project` and `Environment`, `is_hidden` is the only field that can actually be edited.